src/Source/Source.js

```diff
@@ -0,0 +1,104 @@
+import Extent from '../Core/Geographic/Extent';
+
+export const supportedFormats = [
+    'image/png',
+    'image/jpg',
+    'image/jpeg',
+    'image/x-bil;bits=32'];
+
+/**
+ * @typedef {Object} NetworkOptions - Options for fetching resources over the
+ * network. For json or xml fetching, this object is passed as it is to fetch
+ * as the init object, see [fetch documentation]{@link https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch#Parameters}.
+ * @property {string} crossOrigin For textures, only this property is used. Its
+ * value is directly assigned to the crossorigin property of html tags.
+ * @property * Same properties as the init parameter of fetch
+ */
+
+/**
+ * @typedef {object} sourceParams
+ * @property {string} protocol source's protocol (wmts, wms, wfs, file, tms, static)
+ * @property {string} url Base URL of the repository or of the file(s) to load
+ * @property {NetworkOptions} [networkOptions = { crossOrigin: 'anonymous' }] the base url to fetch data source
+ * @property {string} [projection] data's projection
+ * @property {Extent} [extent] data's extent
+ * @property {Attribution} [attribution] Attribution The intellectual property rights for the source
+ * @property {string} [format] data format
+ */
+
+/**
+ * Source are parameters to fetch source
+ *
+ * @class Source
+ */
+class Source {
+    /**
+     * @param  {sourceParams}  source  object to set source
```

Comment:
You should remove the `@class` declaration above, and write a `@constructor` here, to avoid duplication in the generated page of the documentation. 

You should also write an example of declaring a random Source. Best would be to have it for all Sources files.